Example-Swift/PhotosProvider.swift

```diff
@@ -7,39 +7,43 @@
 //
 
 import UIKit
-import NYTPhotoViewer
 
-/**
-*   In Swift 1.2, the following file level constants can be moved inside the class for better encapsulation
-*/
-let CustomEverythingPhotoIndex = 1, DefaultLoadingSpinnerPhotoIndex = 3, NoReferenceViewPhotoIndex = 4
-let PrimaryImageName = "NYTimesBuilding"
-let PlaceholderImageName = "NYTimesBuildingPlaceholder"
+/// A component of your data layer, which might load photos from the cache or network.
+final class PhotosProvider {
+    typealias Slideshow = [Photo]
 
-class PhotosProvider: NSObject {
+    /// Simulate a synchronous fetch of a slideshow, perhaps from a local database.
+    func fetchDemoSlideshow() -> Slideshow {
+        return Array(0...4).map { demoPhoto(identifier: $0) }
+    }
 
-    let photos: [ExamplePhoto] = {
-        
-        var mutablePhotos: [ExamplePhoto] = []
-        var image = UIImage(named: PrimaryImageName)
-        let NumberOfPhotos = 5
-        
-        func shouldSetImageOnIndex(photoIndex: Int) -> Bool {
-            return photoIndex != CustomEverythingPhotoIndex && photoIndex != DefaultLoadingSpinnerPhotoIndex
+    /// Simulate fetching a photo from the network.
+    /// For simplicity in this demo, errors are not simulated, and the callback is invoked on the main queue.
+    func fetchPhoto(named name: String, then completionHandler: @escaping (UIImage) -> Void) {
+        let delay = Double(arc4random_uniform(3) + 2)
+        DispatchQueue.main.asyncAfter(deadline: .now() + delay) {
+            guard let result = UIImage(named: name) else { fatalError("Image '\(name)' could not be loaded from the bundle.") }
+            completionHandler(result)
         }
-        
-        for photoIndex in 0 ..< NumberOfPhotos {
-            let title = NSAttributedString(string: "\(photoIndex + 1)", attributes: [NSForegroundColorAttributeName: UIColor.whiteColor()])
-            
-            let photo = shouldSetImageOnIndex(photoIndex) ? ExamplePhoto(image: image, attributedCaptionTitle: title) : ExamplePhoto(attributedCaptionTitle: title)
-            
-            if photoIndex == CustomEverythingPhotoIndex {
-                photo.placeholderImage = UIImage(named: PlaceholderImageName)
-            }
-            
-            mutablePhotos.append(photo)
+    }
+}
+
+extension PhotosProvider {
+    fileprivate func demoPhoto(identifier: Int) -> Photo {
+        let photoName: String
+        let photoSummary: String
+        let photoCredit: String
+
+        if (arc4random_uniform(2) == 0 && identifier != 0) {
+            photoName = "Chess"
+            photoSummary = "Children gather around a game of chess during the Ann Arbor Summer Festival. (Photo ID \(identifier))"
```

Comment:
In the original example, the photo index was used to populate the caption. Can you do the same here to be consistent? Thanks.